driver-service/src/main/java/com/modsen/driverservice/services/DriverServiceImpl.java

```diff
@@ -0,0 +1,230 @@
+        }
+        Driver driver = repositoryFunc.apply(param)
+                .orElseThrow(() -> new DriverNotFoundException(exMessage));
+        float ratingSum = driver.getAverageRating() * driver.getRatingsCount();
+        int newRatingsCount = driver.getRatingsCount() + 1;
+        return driverMapper.entityToRespDto(driverRepository.save(
+                    driver.setAverageRating((ratingSum + rating) / newRatingsCount)
+                    .setRatingsCount(newRatingsCount)
+            ));
+    }
+
+    private void checkDriverPhoneExist(String phone)
+    {
+        checkDriverParamExist(
+                phone,
+                driverRepository::findByPhone,
+                String.format(ExceptionMessage.DRIVER_PHONE_ALREADY_EXIST_EXCEPTION,phone)
+        );
+    }
+
+    private void preUpdateAllParamsCheck(DriverRequest driverDto, Long id)
+    {
+        preUpdateEmailCheck(id, driverDto);
+        preUpdatePhoneCheck(id, driverDto);
+    }
+
+    private <T> void checkDriverParamExist(T param,
+                                          Function<T, Optional<Driver>> repositoryFunc,
+                                          String exMessage)
+            throws DriverAlreadyExistException{
+        repositoryFunc.apply(param).orElseThrow(() -> new DriverAlreadyExistException(exMessage));
+
+    }
+
+    private void preUpdateEmailCheck(Long id, DriverRequest driverDto)
+    {
+        Driver driver = getDriverOrThrow(id);
+        if (!driver.getEmail().equals(driverDto.getEmail()))
+            checkDriverEmailExist(driverDto.getEmail());
+
+    }
+
+    private void preUpdatePhoneCheck(Long id, DriverRequest driverDto)
+    {
+        Driver driver = getDriverOrThrow(id);
+        if (!driver.getPhone().equals(driverDto.getPhone()))
+            checkDriverPhoneExist(driverDto.getPhone());
+    }
+
+    public DriverResponse setAutoById(Long driver_id, AutoDto autoDto)
+    {
+        return setAuto(
+                driver_id,
+                autoDto,
+                driverRepository::findById,
+                String.format(ExceptionMessage.DRIVER_NOT_FOUND_EXCEPTION, driver_id)
+        );
+    }
+
+    private <T> DriverResponse setAuto(T param,
+                                       AutoDto autoDto,
+                                       Function<T, Optional<Driver>>repositoryFunc,
+                                       String exceptionMessage)
+    {
+        Driver driver = repositoryFunc.apply(param).orElseThrow(() -> new DriverNotFoundException(exceptionMessage));
+        autoRepository.findByNumber(autoDto.getNumber()).orElseThrow(() -> new AutoAlreadyExistException(String.format(
+                ExceptionMessage.AUTO_NUMBER_ALREADY_EXIST_EXCEPTION,
+                autoDto.getNumber()))
+        );
+        if (!driver.getAutos().isEmpty())
+            throw new DriverAlreadyHaveAutoException(ExceptionMessage.DRIVER_ALREADY_HAVE_AUTO_EXCEPTION);
+        else {
```

Comment:
Этот else можно убрать 
```suggestion
       if (!driver.getAutos().isEmpty()) {
            throw new DriverAlreadyHaveAutoException(ExceptionMessage.DRIVER_ALREADY_HAVE_AUTO_EXCEPTION);
        }
        
        driver.getAutos().add(autoMapper.dtoToEntity(autoDto));
        return driverMapper.entityToRespDto(driverRepository.save(driver));
```